opc/resource_lbaas_load_balancer.go

```diff
@@ -0,0 +1,327 @@
+		Delete: resourceOPCLoadBalancerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateLoadBalancerResourceName,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"ip_network": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				ValidateFunc: validateComputeResourceFQDN,
+			},
+			"parent_load_balancer": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"permitted_clients": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"permitted_methods": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"policies": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"scheme": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"INTERNET_FACING",
+					"INTERNAL",
+				}, true),
+			},
+			"server_pool": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateOriginServerPoolURI,
+			},
+			"tags": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			// Read only attributes
+			"balancer_vips": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+			"canonical_host_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cloudgate_capable": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceOPCLoadBalancerCreate(d *schema.ResourceData, meta interface{}) error {
+	lbaasClient, err := meta.(*Client).getLBaaSClient()
+	if err != nil {
+		return err
+	}
+	lbClient := lbaasClient.LoadBalancerClient()
+
+	input := lbaas.CreateLoadBalancerInput{
+		Name:   d.Get("name").(string),
+		Region: d.Get("region").(string),
+	}
+
+	if description, ok := d.GetOk("description"); ok {
+		input.Description = description.(string)
+	}
+
+	if enabled, ok := d.GetOk("enabled"); ok {
+		input.Disabled = getDisabledStateKeyword(enabled.(bool))
+	}
+
+	if ipNetwork, ok := d.GetOk("ip_network"); ok {
+		input.IPNetworkName = ipNetwork.(string)
+	}
+
+	if scheme, ok := d.GetOk("scheme"); ok {
```

Comment:
Since `scheme` is Required, we can just set it when initializing the input

Reply:
done